normandy/recipes/models.py

```diff
@@ -227,6 +196,10 @@ def update_signature(self):
             self.signature = None
             return
 
+        # Don't sign recipe that aren't enabled
+        if not self.enabled:
+            return
```

Comment:
This should probably raise an error instead. I checked how the management command works and it only ever uses `Recipe.objects.only_approved()` so the management command should never get here. 

However, I'm guessing it *could* happen if you use the API to sign a recipe individually and manually and that should trigger some sort of error. But can you? Can you sign an individual recipe with the API?? If so, it should be explicit that the request was "bad".

Reply:
I think eventually raising an error is a good idea, but we aren't quite there yet. I want to see how this shakes out a little before changing the signing logic more. After that, we could make this raise instead of return early.

Also, recipes can be approved but not enabled, so there are plenty of times that an approved recipe should not be signed.

Finally, the API does not expose signing directly. It is done implicitly. Previously basically everything was signed, though after this patch, (hopefully) only recipes that should be going to users will be signed.